npalm/terraform-aws-gitlab-runner
# refactor: Harmonize subnet variables

## Description

Deprecates `subnet_ids_gitlab_runner` and `subnet_id_runners` and introduces `subnet_id` to make the module more readable as explained in #411 and #419. Thus it it no longer possible to place the runner and the executors in different subnets. For a resilient configuration deploy the module once per availability zone.

Closes #419

## Migrations required

No, but migration should be done.

Replace `subnet_ids_gitlab_runner` and `subnet_id_runners` by `subnet_id`. Please notice that it is no longer possible to place the runner and executors in different subnet. For a resilient configuration, I suggest to deploy the module once per availability zone.

## Verification

Checked the following:
- config.toml checked for subnet value using `subnet_id`. Success. Correct subnet used.
- Check that the Runner is created in correct subnet using `subnet_id`. Success.
- Remove `subnet_id` and use the old variables `subnet_ids_gitlab_runner` and `subnet_id_runners` again. `terraform plan` does not show any differences. Success.

## Documentation

We use [pre-commit](https://pre-commit.com/) to update the Terraform inputs and outputs in the documentation via [terraform-docs](https://github.com/terraform-docs/terraform-docs). Ensure you have installed those components.
